src/Console/Command/CompileCommand.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace Castor\Console\Command;
+
+use Castor\PathHelper;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @internal
+ */
+class CompileCommand extends Command
+{
+    private HttpClientInterface $httpClient;
+    private Filesystem $fs;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->httpClient = HttpClient::create();
+        $this->fs = new Filesystem();
+    }
+
+    protected function configure(): void
+    {
+        $this->getDefinition()->addOptions(RepackCommand::getOptions());
+
+        $this
+            ->setName('compile')
+            ->addOption('arch', null, InputOption::VALUE_REQUIRED, 'Target architecture for PHP compilation', 'x86_64', ['x86_64', 'aarch64'])
+            ->addOption('php-version', null, InputOption::VALUE_REQUIRED, 'PHP version in major.minor format', '8.2')
```

Comment:
Since it's packaging a standalone php with a castor app in a single binary, if you're castor app only support a specific version you'll probably want to specify it here.

Reply:
Yes, but because on semver, it's not possible that an app work on 8.1, but not on 8.3, right?

Reply:
There are changes between PHP versions that are not compatible with previous ones:

- https://www.php.net/manual/en/migration83.incompatible.php
- https://www.php.net/manual/en/migration82.incompatible.php

It's like a way out for people who could be affected.

Also, not all PHP extensions work right away with a new PHP version, so users might have to stick with the previous version for a while.

Reply:
mouais, okay, let's keep this option